doc/Dependencies.md

```diff
@@ -0,0 +1,20 @@
+Third Party Dependencies
+========================
+
+## Unaudited Required Dependencies
+
+`libtock-rs` has the following required dependencies, none of which are
+currently audited:
+
+* [`libcore`](https://github.com/rust-lang/rust/tree/master/src/libcore),
+  included as part of the Rust toolchain and implicitly added by the Rust
+  compiler.
+* [`syn`](https://crates.io/crates/syn), pulled in by `libtock_codegen`.
```

Comment:
This is also a dependency of the Rust compiler as far as I can judge.

Reply:
I expanded it to the Rust toolchain in general. I don't think we want to list each individual library (e.g. `compiler-rt`/`compiler-builtins`?), and we don't have to because they're shipped as a unit and maintained by the same group (so the trust model is a constant).

I called out `cargo`, `rustc`, and `libcore` specifically because those are the pieces we use directly.